evm/test/forks/HubMainnetFork.t.sol

```diff
@@ -0,0 +1,373 @@
+  }
+
+  function test_VerifyEvmProposerParams() public view {
+    assertEq(
+      address(hubEvmSpokeAggregateProposer.wormhole()), EXPECTED_WORMHOLE_CORE, "EvmAggProposer wormholeCore mismatch"
+    );
+    assertEq(address(hubEvmSpokeAggregateProposer.HUB_GOVERNOR()), GOV_ADDR, "EvmAggProposer governor mismatch");
+    assertEq(
+      hubEvmSpokeAggregateProposer.maxQueryTimestampOffset(),
+      EXPECTED_MAX_QUERY_OFFSET,
+      "EvmAggProposer maxQueryTimestampOffset mismatch"
+    );
+    // Owner check is in Roles test
+  }
+
+  function test_VerifySolanaDecoderParams() public view {
+    assertEq(
+      address(hubSolanaSpokeVoteDecoder.wormhole()), EXPECTED_WORMHOLE_CORE, "SolanaDecoder wormholeCore mismatch"
+    );
+    assertEq(address(hubSolanaSpokeVoteDecoder.HUB_VOTE_POOL()), HUB_VOTE_POOL_ADDR, "SolanaDecoder target mismatch");
+    assertEq(
+      hubSolanaSpokeVoteDecoder.SOLANA_TOKEN_DECIMALS(),
+      EXPECTED_SOLANA_DECIMALS,
+      "SolanaDecoder tokenDecimals mismatch"
+    );
+    assertEq(address(hubVotePool.voteTypeDecoder(5)), HUB_SOLANA_VOTE_DECODER_ADDR, "SolanaDecoder query type mismatch");
+  }
+
+  // --- Role / Ownership Verification Tests ---
+
+  function test_VerifyTimelockRoles() public view {
+    assertTrue(timelock.hasRole(PROPOSER_ROLE, GOV_ADDR), "Governor lacks PROPOSER_ROLE");
+    assertTrue(timelock.hasRole(EXECUTOR_ROLE, GOV_ADDR), "Governor lacks EXECUTOR_ROLE");
+    assertTrue(timelock.hasRole(CANCELLER_ROLE, GOV_ADDR), "Governor lacks CANCELLER_ROLE");
+    // Check Foundation canceller role (using placeholder address)
+    assertTrue(timelock.hasRole(CANCELLER_ROLE, WORMHOLE_FOUNDATION_ADDR), "Foundation lacks CANCELLER_ROLE"); // Will
+      // fail until placeholder updated & role granted
+    // Check admin role transfers
+    assertFalse(timelock.hasRole(TIMELOCK_ADMIN_ROLE, actualDeployer), "Deployer still has TIMELOCK_ADMIN_ROLE");
+    assertTrue(timelock.hasRole(TIMELOCK_ADMIN_ROLE, TIMELOCK_ADDR), "Timelock lacks TIMELOCK_ADMIN_ROLE");
+  }
+
+  function test_VerifyExtenderRoles() public view {
+    // Verify against the *intended final* admin address (Wormhole Foundation)
+    // NOTE: This will FAIL against current testnet deploy where admin is actualDeployer
+    assertEq(extender.voteExtenderAdmin(), WORMHOLE_FOUNDATION_ADDR, "Extender admin mismatch (Expected Foundation)");
+    // Extender owner (Set to Timelock during Hub deployment)
+    assertEq(extender.owner(), TIMELOCK_ADDR, "Extender owner mismatch");
+  }
+
+  function test_VerifyWhitelistedProposer() public view {
+    // Verify against the *intended final* state (HUB_EVM_AGG_PROPOSER_ADDR)
+    // NOTE: This will FAIL against current testnet deploy where proposer is address(0)
+    // The proposer must be set via a governance action after deployment.
+    assertEq(gov.whitelistedProposer(), address(0), "WhitelistedProposer is set");
+  }
+
+  function test_VerifySpokeRegistrations() public view {
```

Comment:
Can we move hub registration tests to a separate file as we will run a separate script to register the contracts. This will also transfer hub vote pool ownership to the timelock

Reply:
https://github.com/wormhole-foundation/multigov/pull/268/commits/547f119360c239d81adc92bd17eea9f165de25ac